core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -327,6 +328,7 @@ class TransactionStateManager(brokerId: Int,
         }
 
         def shouldInclude(txnMetadata: TransactionMetadata): Boolean = {
+          val now : Long = System.currentTimeMillis()
```

Comment:
Can we move this outside of the callback so that all transactions would be checked against the same timestamp?